### ISSUE TITLE
Code review: SelectByText promises to substring match, but match is on exact text

### ISSUE THREAD

**Issue by Autom8edChaos** [SeleniumHQ/selenium#3575]
While investigating an issue where certain text does not get selected, I encountered the following on the [**dotnet** branch](selenium/dotnet/src/support/UI/SelectElement.cs):

In the currect version, the method `SelectElement.SelectByText(string text)` promises to select the text on substring match:

```
        /// <param name="text">The text of the option to be selected. If an exact match is not found,
        /// this method will perform a substring match.</param>
```

but when I am looking in how the item(s) to select are picked from a list of candidates, the text from a select node is compared to the literal text:

```
   foreach (IWebElement option in candidates)
                {
                    if (text == option.Text)   // <-- literal match
                    {
                        SetSelected(option, true);
//...
```

Do you agree that this is indeed an issue?

**Comment by seanrand57** [comment]
Raised a PR for this fix: https://github.com/SeleniumHQ/selenium/pull/3755

**Comment by jimevans** [comment]
PR applied. Fixed in d0accdf.
